src/app/components/product-table/product-table.component.ts

```diff
@@ -35,12 +35,19 @@ export class ProductTableComponent implements OnInit, AfterViewInit {
     this.searchValue = value;
   }
 
+  constructor(private cdr: ChangeDetectorRef) {}
+
+  isEllipsisActive(e: HTMLElement): boolean {
+    return e ? e.offsetHeight < e.scrollHeight : false;
+  }
+
   ngOnInit(): void {
     this.dataSource.filterPredicate = (value: Product, filter: string) =>
       value.name.toLowerCase().indexOf(filter.toLowerCase()) !== -1;
   }
 
   ngAfterViewInit(): void {
     this.dataSource.sort = this.sort;
+    this.cdr.detectChanges();
```

Comment:
what for do we need this? it should work properly without it.

Reply:
There was this error `ExpressionChangedAfterItHasBeenCheckedError` when I added the function `isEllipsisActive` to dynamically resolve `matTooltipDisabled` prop. It solved the error, but it appears only partially because when I click on the next table page button, the error is logged again. It completely disappears only if you detect changes in ngOnInit

Reply:
I see.
Let's simplify it. let's always show tooltip. So you will no need to make that conditional comparison.